src/Streamly/Internal/Data/Parser/ParserD.hs

```diff
@@ -493,13 +494,34 @@ sliceSepByMax p n = sliceSepBy p . FL.ltake n
 
 -- | See 'Streamly.Internal.Data.Parser.wordBy'.
 --
--- /Unimplemented/
 --
-{-# INLINABLE wordBy #-}
-wordBy ::
-    -- Monad m =>
-    (a -> Bool) -> Fold m a b -> Parser m a b
-wordBy = undefined
+{-# INLINE wordBy #-}
+wordBy :: Monad m => (a -> Bool) -> Fold m a b -> Parser m a b
+wordBy predicate (Fold fstep finitial fextract) = Parser step initial extract
+
+    where
+
+    initial = Left' <$> finitial
+
+    step (Left' s) a =
+        if not (predicate a)
+        then do
+            res <- fstep s a
+            return
+                $ case res of
+                      FL.Partial s1 -> Partial 0 $ Left' s1
+                      FL.Done b -> Done 0 b
+        else do
+            ex <- fextract s
+            return $ Partial 0 $ Right' ex
```

Comment:
We should yield the result as early as possible instead of waiting for all the spaces after it to be over. We can declare done as soon as we see the first space.